zap/src/parser/convert.rs

```diff
@@ -176,28 +210,30 @@ impl<'src> Converter<'src> {
 		let from = evdecl.from;
 		let evty = evdecl.evty;
 		let call = evdecl.call;
-		let data = self.ty(&evdecl.data);
-
-		if let EvType::Unreliable = evty {
-			let (min, max) = data.size(tydecls, &mut HashSet::new());
-			let event_id_size = NumTy::from_f64(1.0, self.evdecls.len() as f64).size();
-
-			// We subtract two for the `inst` array.
-			let max_unreliable_size = 900 - event_id_size - 2;
-
-			if min > max_unreliable_size {
-				self.report(Report::AnalyzeOversizeUnreliable {
-					ev_span: evdecl.span(),
-					ty_span: evdecl.data.span(),
-					max_size: max_unreliable_size,
-					size: min,
-				});
-			} else if !max.is_some_and(|max| max < max_unreliable_size) {
-				self.report(Report::AnalyzePotentiallyOversizeUnreliable {
-					ev_span: evdecl.span(),
-					ty_span: evdecl.data.span(),
-					max_size: max_unreliable_size,
-				});
+		let data = evdecl.data.as_ref().map(|ty| self.ty(ty));
+
+		if let Some(data) = &data {
+			if let EvType::Unreliable = evty {
```

Comment:
```suggestion
		if data.is_some() && evty == EvType::Unreliable {
			let data = data.unwrap();
```

Reply:
I personally think it's less clear this way - what's your intent here?

Reply:
less nesting is usually better, and if we're checking two conditions we shouldn't do it with two if statements